tests/run/cpython_capi.pyx

```diff
@@ -75,3 +76,23 @@ if not hasattr(sys, 'pypy_version_info'):
         Traceback (most recent call last):
         SystemError: ...
     """
+
+def test_list_getref_extend_clear(list):
+    """
+    >>> test_list_getref_extend_clear([1, 2, 3])
+    1
+    []
+    [1, 2]
+    >>> test_list_getref_extend_clear([])  #doctest: +ELLIPSIS
+    Traceback (most recent call last):
+    IndexError: ...
+    """
+
+    first = PyList_GetItemRef(list, 0)
+    print(first)
+
+    PyList_Clear(list)
+    print(list)
+
+    PyList_Extend(list, [1, 2])
+    print(list)
```

Comment:
```suggestion
    1
    [1, 2, 3, 4, 5]
    []
    >>> test_list_getref_extend_clear([])  #doctest: +ELLIPSIS
    Traceback (most recent call last):
    IndexError: ...
    """

    first = PyList_GetItemRef(list, 0)
    print(first)

    PyList_Extend(list, [4, 5])
    print(list)
    
    PyList_Clear(list)
    print(list)
```

I'm just going to swap the extend and clear tests - if something's wrong with extend I think it's more likely to show up on a non-empty list.

Beyond that looks good - I'll set it to merge when it passes.